front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsControllerV2.groovy

```diff
@@ -0,0 +1,163 @@
+public class ApplicationsControllerV2 {
+  @Autowired
+  MessageSource messageSource
+
+  @Autowired
+  ApplicationDAO applicationDAO
+
+  @Autowired
+  ProjectDAO projectDAO
+
+  @Autowired
+  NotificationDAO notificationDAO
+
+  @Autowired
+  PipelineDAO pipelineDAO
+
+  @Autowired
+  PipelineStrategyDAO pipelineStrategyDAO
+
+  @Autowired
+  List<ApplicationValidator> applicationValidators
+
+  @Autowired(required = false)
+  List<ApplicationEventListener> applicationEventListeners = []
+
+  @Autowired
+  Registry registry
+
+  @PreAuthorize("@fiatPermissionEvaluator.storeWholePermission()")
+  @PostFilter("hasPermission(filterObject.name, 'APPLICATION', 'READ')")
+  @ApiOperation(value = "", notes = """Fetch all applications.
+
+    Supports filtering by one or more attributes:
+    - ?email=my@email.com
+    - ?email=my@email.com&name=flex""")
+  @RequestMapping(method = RequestMethod.GET)
+  Set<Application> applications(@RequestParam Map<String, String> params) {
+    if (params.isEmpty()) {
+      return applicationDAO.all()
+    }
+
+    return applicationDAO.search(params)
+  }
+
+  @PreAuthorize("hasPermission(#app.name, 'APPLICATION', 'CREATE')")
+  @ApiOperation(value = "", notes = "Create an application")
+  @RequestMapping(method = RequestMethod.POST, value = "")
+  Application create(@RequestBody final Application app) {
+    return getApplication().initialize(app).withName(app.getName()).save()
+  }
+
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'WRITE')")
+  @ApiOperation(value = "", notes = "Delete an application")
+  @RequestMapping(method = RequestMethod.DELETE, value = "/{application}")
+  void delete(@PathVariable String application, HttpServletResponse response) {
+    getApplication().initialize(new Application().withName(application)).delete()
+    response.setStatus(HttpStatus.ACCEPTED.value())
+  }
+
+  @PreAuthorize("hasPermission(#app.name, 'APPLICATION', 'WRITE')")
+  @ApiOperation(value = "", notes = "Update an existing application")
+  @RequestMapping(method = RequestMethod.PUT, value = "/{name}")
+  Application update(@PathVariable String name, @RequestBody final Application app) {
+    if (name.trim() != app.getName()) {
+      log.error("non matching application name {}", name)
+      throw new InvalidApplicationRequestException("provided application name does not match")
+    }
+
+    def application = getApplication()
+    Application existingApplication = application.findByName(app.getName())
+    application.initialize(existingApplication).withName(app.getName()).update(app)
+    return app
+  }
+
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ')")
+  @ApiOperation(value = "", notes = "Fetch a single application by name")
+  @RequestMapping(method = RequestMethod.GET, value = "/{application}")
+  Application get(@PathVariable final String application) {
+    if (!application?.trim()) {
+      throw new NotFoundException("No application name provided")
```

Comment:
You're gonna want the `.+` syntax; otherwise, Spring is going to incorrectly interpret any application names with a period in them, assuming it's a file extension. This only happens on terminal path variables, i.e. `/applications/{appName}` needs it; /applications/{appName}/foo` does not.

In its current state, a request to `/applications/my.app` will result in the application name being interpreted as `my` instead of `my.app`.